### PR TITLE
🐛 Malformed resolved analytics URL will trigger user error

### PR DESCRIPTION
**Pull request by alanorozco**
After #31594, URLs are parsed using native `URL()`.

Previously, invalid URLs parsed with `<a>` would have been parsed without error, but incorrectly, for example:

```js
parseUrlWithA('https://?foo'); // 'https:/?foo'
new URL('https://?foo'); // throws error
```
 
In this case, #31883 is triggered by `expandUrlAsync` failing with a certain vendor config:

https://github.com/ampproject/amphtml/blob/b445001b99f6ccddc86a6f6e3812a774d4c9ec5a/extensions/amp-analytics/0.1/vendors/mpulse.json#L2-L4

When `beacon_url` fails to resolve, we get the above case:

```js
new URL('https://?h.d=...'); // throws error
```

This can be triggered by any vendor request whose required interpolation values are missing or create an invalid URL, so it's appropriate to redirect them as user errors.

### CODE REVIEW

**Review by micajuine-ho — COMMENTED** [review]
Thanks for the fix @alanorozco! 

If we are fixing this for `baseUrlPromise_`, then we should also catch any URL errors within `requestOriginPromise_` https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/requests.js#L166-L183.

For reference, if [`requestOrigin`](https://amp.dev/documentation/components/amp-analytics/#define-a-request-origin) is defined in the analytics config, we'll extract the origin from it and prepend it to the baseUrl. `requestOrigin` only allows a absolute URL, but since we want amp-analytics to handle malformed urls, I think we do something like:

```

      this.requestOriginPromise_ = this.variableService_
        .expandTemplate(...)
        .then((expandedRequestOrigin) => {
          return this.urlReplacementService_.expandUrlAsync(...);
        })
        .catch((e) => {
           user().info(TAG, "`Could not expand URL "${this.requestOrigin_}"");
        });;
```
as to not stop the execution of amp-analytics, but still handle the malformed URL within amp-analytics. WDYT?

**Comment by alanorozco** [comment]
@micajuine-ho I've consolidated them since they were very similar. PTAL.

File: extensions/amp-analytics/0.1/requests.js [hunk 1]
```diff
@@ -222,6 +200,30 @@ export class RequestHandler {
...
+        this.element_,
+        bindings,
+        this.allowlist_
+      )
+      .then((url) =>
+        this.urlReplacementService_
+          .expandUrlAsync(url, bindings, this.allowlist_)
+          .catch((e) =>
+            userAssert(false, `Could not expand URL "${url}": ${e.message}`)
+          )
```
Comment (micajuine-ho) [inline extensions/amp-analytics/0.1/requests.js:223]: For an empty `requestOrigin`, will parsing with the native `URL()` trigger this catch block and thus stop the request from firing?.

Comment (alanorozco) [reply extensions/amp-analytics/0.1/requests.js:223]: We don't get to this point if `requestOrigin` is empty or unset.

Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/requests.js:223]: SG! Thanks for the confirmation

**Review by micajuine-ho — APPROVED** [review]

**Pull request merged**